integration-libs/opf/checkout/components/b2b/opf-b2b-checkout-payment-type/opf-b2b-checkout-payment-type.component.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  inject,
+  ViewChild,
+} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { PaymentType } from '@spartacus/cart/base/root';
+import {
+  B2BPaymentTypeEnum,
+  CheckoutPaymentTypeFacade,
+} from '@spartacus/checkout/b2b/root';
+import { CheckoutStepService } from '@spartacus/checkout/base/components';
+import { CheckoutStepType } from '@spartacus/checkout/base/root';
+import {
+  getLastValueSync,
+  GlobalMessageService,
+  GlobalMessageType,
+  HttpErrorModel,
+  isNotUndefined,
+  OccHttpErrorType,
+} from '@spartacus/core';
+import {
+  OpfActiveConfiguration,
+  OpfBaseFacade,
+} from '@spartacus/opf/base/root';
+import { BehaviorSubject, combineLatest, Observable, of } from 'rxjs';
+import {
+  catchError,
+  distinctUntilChanged,
+  filter,
+  map,
+  tap,
+} from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-opf-b2b-checkout-payment-type',
+  templateUrl: './opf-b2b-checkout-payment-type.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: false,
+})
+export class OpfB2bCheckoutPaymentTypeComponent {
+  protected opfBaseService = inject(OpfBaseFacade);
+  @ViewChild('poNumber', { static: false })
+  private poNumberInputElement: ElementRef<HTMLInputElement>;
+
+  protected busy$ = new BehaviorSubject<boolean>(false);
+
+  typeSelected?: string;
+  paymentTypesError = false;
+
+  isUpdating$ = combineLatest([
```

Comment:
seeing a number of methods without return type, fee free to add them even though in this it makes sens it is a boolean form method name.

Reply:
This code will be evaluating, but I will add all needed typing for the final version.